dss_datamover/utils/config.py

```diff
@@ -195,9 +196,8 @@ def put(self, subparser):
         subparser.add_argument("--bucket", "-b", type=str, required=False, help='Specify bucket name.. ')
         subparser.add_argument("--prefix", "-p", type=str, required=False,
                                help='Specify object-key prefix, should be <nfs server ip>/<any prefix key>/')
-        subparser.add_argument("--config", "-cfg", type=str, required=False, help='Specify configuration file path')
-        subparser.add_argument("--compaction", "-com", required=False, action='store_true',
-                               help='Enable target compaction')
+        subparser.add_argument("--config", "-cfg", type=str, default='/etc/dss/datamover/config.json', required=False, help='Specify configuration file path')
+        subparser.add_argument("--compaction", "-com", type=str, required=False, help='Enable target compaction')
```

Comment:
add `default=yes` in the options list

Reply:
this does not work, because this default will overwrite anything written in the config file, even if no --compaction flag is passed. Hence, for this option we should keep the default as to whatever is specified in the config file

Reply:
Initial discussion is to have compaction as default irrespective of config file or not. If mentioned in config file as `no` or explicitly through args, then it overrides. Think of backward compatibility as well.

Reply:
Conceptually I agree, mechanically we might need to go a different way about it. Due to the structure of the code, if I set the default in the CLI args it will always override the config file even if "no" is specified in the config file, rendering the config file as useless. Just like we are doing with nfs args, we should rely on config file first then resort to CLI to override. 

If I set the default in the CLI option, then the only time compaction will not run is if compaction arg is used in CLI. Based on these mechanics, should we have the default come from the config or the CLI? If we set the default in CLI then I will remove the option from the config file as it will never be referenced anyways

Reply:
you can think in the simple terms.
If the compaction is present in the config file, then you can take that option. 
Else set it to default value of **yes**
If the user specifies override option, then overwrite that option with his value